File: src/abstractFactory/java/factory/ChampionFactory.java
package abstractFactory.java.factory;

/**
 * Created by kvirus on 2019/6/7 19:41
 * Email @ caoyingkui@pku.edu.cn
 * <p>
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
public interface ChampionFactory {
    public abstract ChampionAbility getAbilities();

    public abstract ChampionConfig getConfig();
}

File: src/abstractFactory/kotlin/factory/ChampionFactory.kt
package abstractFactory.kotlin.factory

/**
 * Created by kvirus on 2019/6/7 19:59
 * Email @ caoyingkui@pku.edu.cn
 *
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */

interface ChampionFactory {
    abstract fun getAbilities(): ChampionAbility
    abstract fun getConfig(): ChampionConfig
}